File: Source/DotNET/Fundamentals/Geospatial/Coordinate.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Geospatial;

/// <summary>
/// Represents a geographic coordinate with longitude and latitude.
/// </summary>
/// <param name="Longitude">The longitude of the coordinate.</param>
/// <param name="Latitude">The latitude of the coordinate.</param>
public record Coordinate(double Longitude, double Latitude);

File: Source/JavaScript/Coordinate.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Point } from './geospatial/Point';

/**
 * @deprecated Use Point from './geospatial' instead. This alias will be removed in a future version.
 * Represents a geographic coordinate with longitude and latitude.
 */
export class Coordinate extends Point {
    /**
     * Converts the Coordinate to a JSON representation (backwards compatible format).
     * @returns {object} JSON representation with longitude and latitude (not GeoJSON).
     */
    toJSON(): object {
        return {
            longitude: this.longitude,
            latitude: this.latitude
        };
    }
}
